C-Unit1/src/E01_Functions.cpp

```diff
@@ -0,0 +1,88 @@
+//
+//  E01_Functions.cpp
+//  Unit1_exercise1_understanding_functions
+//
+
+/*
+ 
+ Note:
+ Implement the functions in this exercise file to make the tests
+ pass in E01_FunctionsSpec.cpp file.
+ 
+ */
+
+
+//
+// Note:
+//  Think of all the edge cases.
+//
+// returns the mid-point of
+// given 2 points on X-axis
+//
+int midPoint(int x1, int x2) {
+    return (unsigned)(x1+x2)/2;
+}
+
+//
+// sum of most significant digits of non-negative integers.
+//
+int sumOfMSDs(int num1, int num2) {
+	int n1=0,n2=0;
+	while (num1 && num2 )
+	{
+		n1 = num1;
+		n2 = num2;
+		num1 /= 10;
+		num2 /= 10;
+	}
+	//now either num1 will be zero or num2 will be zero so from below two loops either of the one will be running 
+	while (num1)
+	{
+		n1 = num1;
+		num1 /= 10;
+	}
+	while (num2)
+	{
+		n2 = num2;
+		num2 /= 10;
+	}
+   return n1+n2;
+}
+
+//
+// sum of two 100 digit positive numbers
+//
+void sumOf100DigitNumbers(int num1[100], int num2[100], int sum[101]) {
+	int i,carry=0,temp;
+	for (i = 99; i > -1; i--)
+	{
+		temp = num1[i] + num2[i] + carry;
+		sum[i + 1] = temp % 10;
+		carry = temp / 10;
+	}
```

Comment:
Did the tests pass for this?

Reply:
Yes all the test cases passes

Reply:
I'll send the screenshot in the evening

Reply:
![image](https://user-images.githubusercontent.com/44928551/66210682-04d67a00-e6d8-11e9-9da7-253bafefe6c4.png)

Reply:
Okay good to go. We'll merge this.